app/assets/javascripts/angular/controllers/query/query_repl_controller.js.es6

```diff
@@ -78,11 +77,25 @@
       this._navigationGuard.disable();
       this._setParameterDefaultValues(this.query.item.version.parameters, this.resultRunner.substitutionValues);
       let result = this.results.collection.shift();
+      if(this._options.skipSave) {
+        /* FIXME: this is for alerts where we want to pop up the repl and save later
+           We probably have to rethink the alerts UI/UX to take this hack out */
+        let copy = new this._Query();
+        copy.internalize(this.query.item);
+        this._modalInstance.close(copy);
+      } else {
+        this.query.save(_.exists(result) ? { result_id: result.item.id } : {})
+          .then(item => {
+            let model = new this._Query();
+            model.internalize(item);
+            this._modalInstance.close(model);
```

Comment:
is it worth methodizing this little routine, since it is used twice, and a name like `close_and_return_query` might make it super clear what;s going on?